packages/astro/src/core/compile/compile.ts

```diff
@@ -46,6 +46,7 @@ export async function compile({
 			scopedStyleStrategy: astroConfig.scopedStyleStrategy,
 			resultScopedSlot: true,
 			transitionsAnimationURL: 'astro/components/viewtransitions.css',
+			annotateSourceFile: viteConfig.mode === 'development' && astroConfig.experimental.devOverlay,
```

Comment:
I think it would be better to test with something like:

```suggestion
			annotateSourceFile: viteConfig.command === 'serve' && astroConfig.experimental.devOverlay,
```

To check if we're running in dev mode. `mode` in Vite is used to refer to the kind of dev or build, e.g. it could be `staging`.

Reply:
Thank you! I got confused because this function doesn't take the usual AstroSettings object